py/modbuiltins.c

```diff
@@ -460,8 +460,33 @@ MP_DEFINE_CONST_FUN_OBJ_1(mp_builtin_repr_obj, mp_builtin_repr);
 STATIC mp_obj_t mp_builtin_round(size_t n_args, const mp_obj_t *args) {
     mp_obj_t o_in = args[0];
     if (MP_OBJ_IS_INT(o_in)) {
-        return o_in;
-    }
+        if(n_args == 1 || mp_obj_is_true(mp_binary_op(MP_BINARY_OP_MORE_EQUAL, args[1], mp_obj_new_int(0)))){
+            return o_in;
+        }
+        if (!MP_OBJ_IS_INT(args[1])){
+            mp_raise_TypeError("Second parameter must be an integer");
+        }
```

Comment:
Instead of doing this check for int, just extract the int like `mp_int_t num_dig = mp_obj_get_int(args[1]);`.  That will raise an exception if the arg is not an int.  Then you'll also have the int value as a machine int which will make the rest of the code simpler (it won't allow bignum for this arg but it's unlikely that one will ever pass huge values to the number of digits).